### ISSUE TITLE
Could not resolve "firebase/storage"

### ISSUE THREAD

**Issue by mathias-mueller** [lion5/component-library#97]
The library apparently needs the "firebase" dependency.

Currently I am only using the "LeafletMap" component.

```html
<script setup lang="ts">
import {LeafletMap} from "@lion5/component-library";</script>

<template>
    <LeafletMap
        id="map"
        :initial-center="center.toLatLng()"
        :initial-zoom="5"
    />
</template>
<script lang="ts">
import {GpsLocation} from "@lion5/component-library";
import {computed} from "vue";

const center = computed(
    (): GpsLocation => new GpsLocation(49.8857, 10.8936)
)

</script>
```

The error:
```
[ERROR] Could not resolve "firebase/storage"

 

    node_modules/@lion5/component-library/dist/library.es.js:7:85:
      7 │ ... { getStorage as v0, ref as gl, uploadBytes as y0, getDownloadURL as b0 } from "firebase/storage";
        ╵                                                                                   ~~~~~~~~~~~~~~~~~~

 

  You can mark the path "firebase/storage" as external to exclude it from the bundle, which will
  remove this error.
```

**Comment by michigg** [comment]
I've looked at various build tools and other options for our problem:
- https://lerna.js.org/
- https://nx.dev/
- https://rushjs.io/
- https://turbo.build/
- https://bit.dev/
- Workspaces
	- https://docs.npmjs.com/cli/v7/using-npm/workspaces
	- https://classic.yarnpkg.com/en/docs/workspaces/
	- https://pnpm.io/workspaces

Currently I try `lerna`. It seems to be the oldest on market and still supported. It is backed by the nx and get a lot of update in the last time. It has also a graph preview (the preview from nx) to display the dependencies between the packages. See #98 for the latest updates on that topic.

**Comment by nagychris** [comment]
> The `firebase/storage` and other external libs we defined as optional are imported at the top of the bundled js file. So if this library is used in other projects it requires all optional dependencies being loaded.
> 
> Quick fix: install all optional packages or define these packages as external
> 
> I see two options for a future fix. The first option - maybe more easier - is to convert all imports of the optional dependencies to dynamic imports. Using dynamic imports we need to convert all affected code parts to asynchronous code.
> 
> We should divide the package into smaller specialized sub packages. For this we need to introduce multi package bundling. The new packages can be:
> 
> * `@lion5/component-library-shared`
>   
>   * contains shared code between the libs, like css color, shared libs
> * `@lion5/component-library-core`:
>   
>   * components that do only need contents for `@lion5/component-library-shared`  here
>   * should be as small as possible, no additional npm dependencies
> * `@lion5/component-library-map`:
>   
>   * leaflet related stuff like `LeafletMap`, `GeolocationInput` or `useLeafletMarker`
> * `@lion5/component-library-firebase`
>   
>   * firebase related stuff like `useFirebaseStorage`
> 
> @nagychris what do u think?

I think splitting the component-library package into smaller packages is a good idea to solve this issue. It would also be beneficial to separate the "big" library with many functionalities into individual use-cases. Usually, a library / npm package that you import only fulfils one purpose, to keep your code organized and to minimize the bundle size.
As the component-library currently supports several use cases, I think this separation should be a valid way to make it more usable and to require only the dependencies you really need in your project. 

As you already mentioned, [lerna](https://lerna.js.org/) looks like a good fit for us, as it is also used by other big libraries like Jest. But maybe we should also take a closer look at [turbo](https://github.com/vercel/turbo), as it is used by big companies, and it might be quicker than lerna as its written in Rust/Go.

**Comment by michigg** [comment]
Should be fixed now since #98. @mathias-mueller can you retest it?

**Comment by michigg** [comment]
Mathias tested it. The Problem is solved.
